packages/curve-ui-kit/src/shared/ui/Balance.tsx

```diff
@@ -82,66 +67,54 @@ const BalanceText = <T extends Amount>({ symbol, balance, loading = false }: Bal
 export type Props<T> = {
   /** The token symbol to display */
   symbol: string
-  /** Controls how the max value is displayed:
-   * - 'balance': Shows the balance as a clickable link
-   * - 'button': Shows a separate 'Max' button
-   * - 'off': No max functionality is shown
-   */
-  max: 'balance' | 'button' | 'off'
+  /** Whether the balance is clickable or not */
+  clickable?: boolean
   /** The token balance amount (optional, in case of loading) */
   balance?: T
   /** The USD value of the balance (optional) */
   notionalValueUsd?: T | number
   /** Whether to hide the wallet icon */
   hideIcon?: boolean
-  sx?: SxProps
-  /**
-   * Callback function when max button/balance is clicked.
-   * When using LargeTokenInput, onChange will be called with the max value before this.
-   **/
-  onMax?: () => void
   /** Whether the balance is loading */
   loading?: boolean
-  /** Whether the max button is disabled */
+  /** Whether the clickable balance is disabled (something might be loading?) */
   disabled?: boolean
-  /** Optional test ID for the button */
-  maxTestId?: string
+  sx?: SxProps
+  /** Optional test ID for the clickable balance button */
+  clickTestId?: string
+  /** Callback function when balance is clicked (if enabled). */
+  onClick?: () => void
 }
 
 export const Balance = <T extends Amount>({
   symbol,
-  max,
+  clickable,
   loading = false,
   balance,
   notionalValueUsd,
   hideIcon,
   sx,
-  onMax,
+  onClick,
   disabled,
-  maxTestId,
+  clickTestId,
 }: Props<T>) => (
   <Stack direction="row" gap={Spacing.xs} alignItems="center" sx={sx}>
     {!hideIcon && <AccountBalanceWalletOutlinedIcon sx={{ width: IconSize.xs, height: IconSize.xs }} />}
 
-    {max === 'balance' && balance != null ? (
-      <MaxButton underline onClick={onMax} loading={loading} testId={maxTestId}>
-        <BalanceText symbol={symbol} balance={balance} loading={loading} />
-      </MaxButton>
-    ) : (
+    <WithWrapper
+      Wrapper={BalanceButton}
+      wrap={!!clickable && balance != null}
```

Comment:
`wrap` could just be typed as `unknown` tbh
```suggestion
      wrap={clickable && balance != null}
```